src/bench/readblock.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+#include <bench/data.h>
+
+#include <chainparams.h>
+#include <consensus/validation.h>
+#include <node/blockstorage.h>
+#include <streams.h>
+#include <test/util/setup_common.h>
+#include <util/system.h>
+#include <validation.h>
+
+using node::BLOCK_SERIALIZATION_HEADER_SIZE;
+using node::OpenBlockFile;
+using node::ReadBlockFromDisk;
+using node::ReadRawBlockFromDisk;
+
+static void WriteBlockToDisk()
```

Comment:
It's a little different. Here we are using the `OpenBlockFile` and `FlatFilePos` abstractions so that it will work with `ReadBlockFromDisk` and `ReadRawBlockFromDisk`. If these are internally changed then this benchmark will still work (in theory). In `load_external` it's writing to an arbitrary file that's passed into `LoadExternalBlockFile` which doesn't use these abstractions internally. It's also filling up the file with repeated blocks, whereas here we only write one.

Reply:
It's actually reimplimenting https://github.com/bitcoin/bitcoin/blob/master/src/node/blockstorage.cpp#L670-L691

However, it's private in `blockstorage.cpp`, but might be worth refactoring it to expose it for this benchmark?

Reply:
Updated to use `BlockManager::SaveBlockToDisk`.